tools/serverpod_cli/lib/src/generator/dart/library_generators/entities_library_generator.dart

```diff
@@ -694,7 +740,104 @@ class SerializableEntityLibraryGenerator {
                               config: config,
                             )
                           ]
-                        : [])).call([literalString(field.name)]).code));
+                        : []))));
+              } else if (field.relation is ObjectRelationDefinition) {
+                // Complex relation fields
+                var objectRelation = field.relation as ObjectRelationDefinition;
+                var foreignRelationField = fields
+                    .cast<SerializableEntityFieldDefinition?>()
+                    .firstWhere(
+                        (f) =>
+                            f?.name == objectRelation.fieldName &&
+                            f?.relation is ForeignRelationDefinition,
+                        orElse: () => null);
+
+                if (foreignRelationField == null) {
+                  log.error(
+                      'Could not find foreign relation field ${objectRelation.fieldName} for ${field.name} in ${classDefinition.className}.');
+                  continue;
+                }
+
+                var foreignRelation =
+                    foreignRelationField.relation as ForeignRelationDefinition;
+
+                // Add internal nullable table field
```

Comment:
If it is OK with you I would like to do a full pass on this class in a separate PR. I'm eager to get started with the work but think it would be best in an isolated PR.